diff --git a/sites/all/modules/user_import/user_import.import.inc b/sites/all/modules/user_import/user_import.import.inc
index 2f3ed43..fc9fdc8 100644
--- a/sites/all/modules/user_import/user_import.import.inc
+++ b/sites/all/modules/user_import/user_import.import.inc
@@ -15,6 +15,8 @@ function _user_import_process($settings) {
 	$username_abbreviate = array();
 	$first_line_skip = 0;
 
+	$delimiter = isset($settings['delimiter']) && !empty($settings['delimiter']) ? $settings['delimiter'] : ',';
+
 	ini_set('auto_detect_line_endings', TRUE);
 	$handle = @fopen($settings['filepath'], "r");
 
@@ -24,8 +26,8 @@ function _user_import_process($settings) {
 	// start count of imports on this cron run
 	$processed_counter = 0;
    
-  while ($data = fgetcsv($handle, $line_max, ',')) {
-    
+  while ($data = fgetcsv($handle, $line_max, $delimiter)) {
+
 		$errors = user_import_errors(FALSE, TRUE);
 
 		// if importing, check we are not over max number of imports per cron
